src/graphql.rs

```diff
@@ -729,6 +729,40 @@ fn get_trend(
     signal::filter::filtfilt(&b, &a, &original)
 }
 
+/// Checks after, before, first, and last to see if they follow GraphQL guidelines.
+///
+/// # Errors
+///
+/// Returns an error if the parameters are invalid.
+#[allow(clippy::type_complexity)]
+pub fn validate_pagination_params_and_set_default(
+    after: Option<String>,
+    before: Option<String>,
+    mut first: Option<i32>,
+    mut last: Option<i32>,
+) -> Result<(Option<String>, Option<String>, Option<i32>, Option<i32>)> {
+    match (
+        first.is_some(),
+        last.is_some(),
+        before.is_some(),
+        after.is_some(),
+    ) {
+        (true, true, _, _) => Err("cannot provide both `first` and `last`".into()),
+        (_, _, true, true) => Err("cannot provide both `before` and `after`".into()),
+        (true, _, true, _) => Err("cannot provide both `first` and `before`".into()),
+        (_, true, _, true) => Err("cannot provide both `last` and `after`".into()),
+        (false, false, false, _) => {
+            first = Some(DEFAULT_CONNECTION_SIZE.to_i32().unwrap_or_default());
+            Ok((after, before, first, last))
+        }
+        (false, false, true, false) => {
+            last = Some(DEFAULT_CONNECTION_SIZE.to_i32().unwrap_or_default());
+            Ok((after, before, first, last))
+        }
+        _ => Ok((after, before, first, last)),
+    }
```

Comment:
it seems like we can do this as well.

```

    let after = if let Some(after) = after {
        Some(decode_cursor(&after).ok_or("invalid cursor `after`")?)
    } else {
        None
    };
    let before = if let Some(before) = before {
        Some(decode_cursor(&before).ok_or("invalid cursor `before`")?)
    } else {
        None
    };
    ```

Reply:
Ahh I didn't see that thank you!

Reply:
It should not exist inside match. Please fix.